pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -163,12 +163,17 @@ protected boolean isConsumersExceededOnSubscription() {
         return isConsumersExceededOnSubscription(topic, consumerList.size());
     }
 
+    private void removeAllConsumer(Consumer consumer){
+        while(consumerList.remove(consumer)){
+        }
+    }
+    
     @Override
     public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceException {
         // decrement unack-message count for removed consumer
         addUnAckedMessages(-consumer.getUnackedMessages());
         if (consumerSet.removeAll(consumer) == 1) {
-            consumerList.remove(consumer);
+            removeAllConsumer(consumer);
```

Comment:
The `removeAllConsumer` doesn't look great. I wonder if using `.removeIf` would be better:
```suggestion
            consumerList.removeIf(consumer::equals);
```

Reply:
I agree with this sentiment. I wonder about the data structure itself, too. Why is it a list and not a set?

Reply:
+1 to @lhotari 's suggestion

Reply:
I agree

Reply:
Fixed:
1.do a check in addConsumer;
2.use removeIf;

PTAL,thanks! @lhotari

Reply:
@lordcheng10 there's a minor checkstyle issue in the recent changes. please check that

Reply:
Fixed.
PTAL,thanks! @lhotari